src/docfx/lib/github/GitHubUserCache.cs

```diff
@@ -154,55 +149,59 @@ async Task<(Error, GitHubUser)> GetByCommitCore()
                 Log.Write($"Calling GitHub commit API to resolve {authorEmail}");
                 Telemetry.TrackCacheMissCount(TelemetryName.GitHubUserCache);
 
-                var (error, user) = await _getUserByCommitFromGitHub(repoOwner, repoName, commitSha);
+                var (error, users) = await _getUsersByCommitFromGitHub(repoOwner, repoName, commitSha);
 
                 // When GetUserByCommit failed, it could either the commit is not found or the user is not found,
                 // only mark the email as invalid when the user is not found
-                if (user != null)
+                if (users != null)
                 {
-                    UpdateUser(user);
+                    UpdateUsers(users);
                 }
-                return (error, user);
+
+                return (error, _usersByEmail.TryGetValue(authorEmail, out var user) && user.IsValid() ? user : null);
```

Comment:
will that cause any differences  between `get by commit` and `get by email`?

Reply:
The `string.IsNullOrEmpty(repoOwner)` will make `get by email` return early, so there is no different. But it is bad to mix these into one method, I need to split them in another PR later.